relay/pkg/block/push.go

```diff
@@ -0,0 +1,214 @@
+package block
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/keep-network/tbtc/relay/pkg/btc"
+)
+
+// pullHeadersFromQueue waits until we have `headersBatchSize` headers from
+// the queue or until the queue fails to yield a header for
+// `headerTimeout` duration and returns them from the function.
```

Comment:
Let's clarify the case when the timeout is hit. I was confused when reviewing the code using this function thinking no headers (empty slice) can be returned in the case of a timeout. My proposition:
```
// pullHeadersFromQueue blocks until there is `headersBatchSize` headers in
// the queue or until `headerTimeout` is hit and no more headers are available
// in the queue. Function returns headers from the queue but not less than one
// and no more than `headersBatchSize` headers.
```

Reply:
Sure: https://github.com/keep-network/tbtc/commit/49448a1e7de6aece3ab985d40829ddfbed1346d1 . Keep in mind this function can return an empty slice if the context is cancelled. Flagged that in the comment.